File: 463-island-perimeter/463-island-perimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        count = 0
        for i in range(len(grid)):
            for j in range(len(grid[i])):
                if grid[i][j] == 1:
                    if j-1 < 0 : count += 1
                    if j+1 >= len(grid[i]) : count += 1
                    if i-1 < 0 : count += 1
                    if i+1 >= len(grid) : count += 1
                    if j-1 >= 0 and grid[i][j-1] == 0 : count += 1
                    if j+1 < len(grid[i]) and grid[i][j+1] == 0 : count += 1
                    if i-1 >=0 and grid[i-1][j] == 0 : count += 1
                    if i+1 < len(grid) and grid[i+1][j] == 0: count += 1
                    
                        
        return count 
                

File: 463-island-perimeter/463-island-perimeter.go
func islandPerimeter(grid [][]int) int {
    count := 0
    for i:=0;i<len(grid);i++{
        for j:=0;j<len(grid[i]);j++{
            if grid[i][j] == 1{
                count += 4
                if i>0 && grid[i-1][j] == 1{
                    count -= 2
                }
                if j>0 && grid[i][j-1] == 1{
                    count -= 2
                }
            }
            
        }
    }
    return count
}